openedx/adg/admin_dashboard/admin_settings/views.py

```diff
@@ -59,6 +87,26 @@ class PendingInvitationListView(EnsureAdminUserAuthenticatedMixin, ListView):
     paginate_by = ADMIN_SETTINGS_PAGE_SIZE
     context_object_name = 'invitations'
 
+    def get_context_data(self, *, object_list=None, **kwargs):
+        """
+        Updates list of objects to list of dicts containing the invitations data.
+        """
+        context = super().get_context_data(object_list=object_list, **kwargs)
+
+        context['invitations'] = [
+            {
+                'id': invitation.id,
+                'email': invitation.email,
+                'permission': dict(Invitation.PERMISSION_CHOICES).get(invitation.permission, '-'),
+                'business_unit_title': (
+                    invitation.business_unit.title
+                    if invitation.permission != Invitation.ADG_ADMIN and invitation.business_unit else '-'
+                )
+            } for invitation in context['invitations']
```

Comment:
Wouldn't it be better if we encapsulate this logic in another method and call that method here to get this data?

Reply:
Why extra function calls? We are updating the context so it can go here and also there is no other logic.

Reply:
The whole purpose of `get_context_data` is to create and return context, if we move this logic to another function then `get_context_data` would just be calling that function with no significance if its own existence, so I think this structure is fine.